programs/clearing_house/src/math/margin.rs

```diff
@@ -98,24 +267,21 @@ pub fn calculate_margin_requirement_and_total_collateral(
             "AMM must be updated in a prior instruction within same slot"
         )?;
 
-        let worst_case_base_asset_amount = market_position.worst_case_base_asset_amount()?;
-        let worse_case_base_asset_value =
-            calculate_base_asset_value(worst_case_base_asset_amount, &market.amm, false)?;
-
-        let position_unrealized_pnl = calculate_position_pnl(market_position, &market.amm, false)?;
+        let oracle_price_data = oracle_map.get_price_data(&market.amm.oracle)?;
 
-        let margin_ratio = market.get_margin_ratio(margin_requirement_type);
+        let (perp_margin_requirement, perp_unrealized_pnl) = calculate_perp_equity_value(
```

Comment:
maybe just call this `perp_pnl` since it has unsettled and unrealized?

Reply:
also we might just want to return the `perp_margin_requirement` in quote precision and add it to margin_requirement immediately as well